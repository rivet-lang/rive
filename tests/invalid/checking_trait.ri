trait ToStr {
    fn to_str1(&self) string;
}

fn my_print(v: ToStr) {
    _ = v.to_str1();
}

extend string : ToStr {
    fn to_str1(self) string {
        return self;
    }
}

extend bool : ToStr {
    fn to_str1(&self) string {
        return if self.* { "true" } else { "false" };
    }
}

fn main() {
    my_print(""); // OK
    my_print(true); // OK
    my_print(1); // FAIL
}
