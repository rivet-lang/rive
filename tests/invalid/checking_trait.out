tests/invalid/checking_trait.ri:24:14: error: type `i32` does not implement trait `ToStr`
  24 |     my_print(1); // FAIL
     |              ^
     = note: in argument `v` of function `my_print`
rivetc: error: could not compile module `checking_trait`, aborting due to previous error